tests/unit/test_charm.py

```diff
@@ -6,452 +6,335 @@
-        with self._patch_exec():
-            self.harness.container_pebble_ready("discourse")
-
-        self.assertEqual(
-            self.harness.model.unit.status,
-            BlockedStatus("'saml_sync_groups' cannot be specified without a 'saml_target_url'"),
-        )
-
-    def test_config_changed_when_saml_target_url_and_force_https_disabled(self):
-        """
-        arrange: given a deployed discourse charm with all the required relations
-        act: when saml_target_url configuration is provided and force_https is False
-        assert: it will get to blocked status waiting for the latter.
-        """
-        self.harness.update_config({"saml_target_url": "group1", "force_https": False})
-        with self._patch_exec():
-            self.harness.container_pebble_ready("discourse")
-
-        self.assertEqual(
-            self.harness.model.unit.status,
-            BlockedStatus(
-                "'saml_target_url' cannot be specified without 'force_https' being true"
-            ),
-        )
-
-    def test_config_changed_when_no_cors(self):
-        """
-        arrange: given a deployed discourse charm with all the required relations
-        act: when cors_origin configuration is empty
-        assert: it will get to blocked status waiting for it.
-        """
-        self.harness.update_config({"cors_origin": ""})
-        with self._patch_exec():
-            self.harness.container_pebble_ready("discourse")
-
-        self.assertNotEqual(
-            self.harness.charm._database.get_relation_data(),
-            None,
-            "database name should be set after relation joined",
-        )
-
-        self.assertEqual(
-            self.harness.charm._database.get_relation_data().get("POSTGRES_DB"),
-            "discourse",
-            "database name should be set after relation joined",
-        )
-
-    def test_config_changed_when_throttle_mode_invalid(self):
-        """
-        arrange: given a deployed discourse charm with all the required relations
-        act: when throttle_level configuration is invalid
-        assert: it will get to blocked status waiting for a valid value to be provided.
-        """
-        self.harness.update_config({"throttle_level": "Scream"})
-
-        self.assertEqual(self.harness.model.unit.status.name, BLOCKED_STATUS)
-        self.assertTrue("none permissive strict" in self.harness.model.unit.status.message)
-
-    def test_config_changed_when_s3_and_no_bucket_invalid(self):
-        """
-        arrange: given a deployed discourse charm with all the required relations
-        act: when s3_enabled configuration is True and there's no s3_bucket
-        assert: it will get to blocked status waiting for the latter.
-        """
-        self.harness.update_config(
+@pytest.mark.parametrize(
+    "with_postgres, with_redis, with_ingress, status",
+    [
+        (False, False, False, WaitingStatus("Waiting for database relation")),
```

Comment:
Do you think it would be worth it to use `pytest.param` and use the `id` argument to describe what the case is?

Reply:
Added